src/librustc/ty/context.rs

```diff
@@ -2993,6 +2991,12 @@ impl<T, R, E> InternIteratorElement<T, R> for Result<T, E> {
     }
 }
 
+// We are comparing types with different invariant lifetimes, so `ptr::eq`
+// won't work for us.
+fn ptr_eq<T, U>(t: *const T, u: *const U) -> bool {
```

Comment:
does the necessity of adding this function (and calling through it) perhaps indicate that we should do a crater run on this change?

Reply:
or I must have misunderstood. Am I overestimating the impact of this change?

... this change should only meant to affect MIR-borrowck ... so... did all of these `ptr::eq` calls really only get introduced after we had already moved `rustc` to NLL? I guess I can actually believe that.

Reply:
Or oh, this is *solely* an issue with `#![feature(nll)]`, **not** with borrowck=migrate?

Reply:
yes, okay; privately confirmed that the bug here does not affect borrowck=migrate, and therefore is not a problem for the 2018 edition.